cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -95,4 +110,38 @@ private String exceptionToJson(Exception ex) {
     BasicThrowable basicThrowable = new BasicThrowable(ex);
     return GSON.toJson(basicThrowable);
   }
+
+  private static class RunnableTaskBodyProducer extends BodyProducer {
+    private final InputStream responseStream;
+    private final Consumer<String> stopper;
+    private final String className;
+
+    RunnableTaskBodyProducer(InputStream responseStream, Consumer<String> stopper, String className) {
+      this.responseStream = responseStream;
+      this.stopper = stopper;
+      this.className = className;
+    }
+
+    @Override
+    public ByteBuf nextChunk() throws Exception {
+      byte[] buf = new byte[64 * 1024];
```

Comment:
Why do you do this double copying through the ByteArrayInputStream? Can you just pass the original response and  wrap it into ByteBuffer?

Reply:
Great suggestion. Done!